### DISCUSSION TITLE
이번 V2 이동UI를 사용하고 움직이는 오브젝트를 올라타면 점프에 버그가 생기네요

### DISCUSSION THREAD

**Discussion by ted002** [naverz/zepeto-studio-kor#608] [Issue Report (Legacy)]
https://www.youtube.com/watch?v=icCyrYFkYBg&ab_channel=RobloxMakeFM

해당 영상에서 나온것처럼, 이번에 업데이트된 V2 이동UI를 사용하면 이동하는 오브젝트에서 낙하 애니메이션이 나옵니다. 아래로 내려가는 움직일때는 멀쩡한데, 그 이외의 모든 움직이는 오브젝트에서는 낙하 애니메이션이 반복적으로 재생되며 점프가 아예 불가하게 됩니다.

`import { CharacterController, GameObject, Time, Vector3 } from 'UnityEngine'
import { ZepetoPlayer, ZepetoPlayers } from 'ZEPETO.Character.Controller';
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'
import MoveWithPlat from './MoveWithPlat';

export default class LerpPlatform extends ZepetoScriptBehaviour {
    public AllPosition : Vector3[];

    public Detector:GameObject;
    public Speed:float;

    private m_bToEnd : bool = false;
    private m_bToStart : bool = true;
    private m_bChange : bool = false;
    private m_bStart : bool = false;

    private m_fWaitTime : float = 0;

    public m_fDelayTime : float;
    public m_fStartDelayTime : float;
    public m_iNowPos : int = 0;

    Start() {    
        if(this.m_fStartDelayTime == 0)
        {
            this.m_bStart = true;
        }

    }

    Update()
    {
        if(this.m_bStart)
        {
            if(!this.m_bChange)
            {
                if(this.Detector.GetComponent<MoveWithPlat>().m_bPlayerOn && !this.m_bChange)
                {
                    var PlayerPosition : Vector3 = Vector3.op_Addition(Vector3.MoveTowards(this.transform.position,this.AllPosition[this.m_iNowPos],this.Speed * Time.deltaTime),Vector3.op_UnaryNegation(this.transform.position));

                    ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.GetComponent<CharacterController>().Move(PlayerPosition);
                }

                this.transform.position = Vector3.MoveTowards(this.transform.position,this.AllPosition[this.m_iNowPos],this.Speed * Time.deltaTime);

                if (Vector3.op_Equality(this.transform.position,this.AllPosition[this.m_iNowPos]))
                {
                    this.m_bChange = true;
                    this.m_iNowPos++;
                    if(this.m_iNowPos >= this.AllPosition.length)
                    {
                        this.m_iNowPos = 0;
                    }
                }
            }

            if(this.m_bChange)
            {

                if(this.m_fWaitTime > this.m_fDelayTime)
                {
                    if(this.m_bToStart && !this.m_bToEnd && this.m_bChange)
                    {
                        this.m_bToStart = false;
                        this.m_bToEnd = true;
                    }

                    else if(this.m_bToEnd && !this.m_bToStart && this.m_bChange)
                    {
                        this.m_bToStart = true;
                        this.m_bToEnd = false;
                    }

                    this.m_bChange = false;
                    this.m_fWaitTime = 0;
                }
                else
                {
                    this.m_fWaitTime += Time.deltaTime;
                }
            }
        }
        else
        {
            if(this.m_fWaitTime > this.m_fStartDelayTime)
            {
                this.m_bStart = true;
                this.m_fWaitTime = 0;
            }
            else
            {
                this.m_fWaitTime += Time.deltaTime;
            }

        } 
    }

}`

코드 전문을 함께 드립니다.
MoveWithPlat 이라는 오브젝트를 추가로 붙여서 플레이어가 플랫폼 위에 함께하고 있는지 파악합니다.

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, UseLandingRoll 체크 여부를 알려주시겠어요? 
참고로 UseLandingRoll 사용 여부를 체크하면 캐릭터 낙하속도(Vertical Velocity) 값이 18보다 클 때 발동하는 로직입니다.

**Reply by ted002** [reply]
넵 사용하고 있습니다!

**Reply by Zepeto-TechEvangelist** [reply]
네, 그렇다면 UseLandingRoll 체크를 끈 후에도 동일 증상이 나타나는지 확인 부탁 드립니다.

**Reply by ted002** [reply]
UseLandingRoll을 꺼도 동일한 문제가 발생합니다!

**Reply by Zepeto-TechEvangelist** [reply]
넵, 제보 감사 드립니다.
해당 내용 원인이 파악되어 수정 작업 진행 중입니다. 🙏

**Reply by Zepeto-TechEvangelist** [reply]
월드 1.6.0 버전에서 수정 완료되었습니다 :)
